### ISSUE TITLE
Fairness metrics dashboard with CI integration

### ISSUE THREAD

**Issue by dreamlessx** [dreamlessx/LandmarkDiff-public#48]
## Motivation

Issue #29 covers the research side of the fairness audit (building balanced test sets, running the full evaluation, etc.). This issue is about the engineering side: once we have those metrics, how do we track them over time and catch regressions automatically?

Right now if someone submits a PR that tweaks the deformation vectors or changes the ControlNet conditioning, there is no automated check that performance across Fitzpatrick types stays balanced. We need to catch that before it lands in main.

### 1. Fairness report generator

A script (or extension of `scripts/run_evaluation.py`) that:

- Takes evaluation results JSON as input
- Computes per-group metrics: FID, LPIPS, SSIM, NME, identity similarity
- Groups by Fitzpatrick type (I-VI), using the ITA-based classifier we already have in `evaluation.py`
- Computes the max performance gap between any two groups for each metric
- Outputs a summary table and pass/fail status based on configurable thresholds

Example threshold config:
```yaml
fairness_thresholds:
  max_fid_gap: 15.0          # max FID difference between best and worst group
  max_lpips_gap: 0.05        # max LPIPS difference
  max_failure_rate_gap: 0.10  # max difference in failure rate (identity_sim < 0.6)
```

### 2. GitHub Actions integration

Add a CI job that runs on PRs touching `landmarkdiff/manipulation.py`, `landmarkdiff/conditioning.py`, or model configs:

- Run inference on a small balanced test set (maybe 10 images per Fitzpatrick type, 60 total)
- Generate the fairness report
- Post the summary table as a PR comment
- Fail the check if any threshold is exceeded

This would use a lightweight evaluation (skip FID, just do LPIPS + identity sim) to keep CI times under 15 minutes on a T4 runner.

### 3. Historical tracking

Store fairness metrics per commit in a JSON file (or SQLite db) so we can plot trends over time. The existing `scripts/training_dashboard.py` could be extended to show a fairness tab alongside training loss curves.

## Non-goals

- This is not about building the balanced test set (that is #29)
- Not about adding new demographic dimensions beyond Fitzpatrick types (also #29)
- Not about the clinical review process

## References

- `landmarkdiff/evaluation.py`, `classify_fitzpatrick_ita()` already classifies skin types
- `landmarkdiff/audit.py`, existing audit infrastructure we can build on
- `scripts/run_evaluation.py`, current eval entry point
